doc/aaa/radius_authentication.md

```diff
@@ -0,0 +1,634 @@
+| 1-14           | remote_user    |65534 |65534 | users          | /bin/rbash|
+
+
+A different mapping of MPL can be configured in the /etc/radius_nss.conf,
+on the switch. For example, if the following lines were added to the
+configuration.
+
+```
+user_priv=15;pw_info=remote_user_su;uid=1000;gid=1000;group=sudo,docker;dir=/home/admin;shell=/bin/bash
+user_priv=7;pw_info=netops;uid=2007;gid=100;group=users;dir=/home/netops;shell=/bin/rbash
+user_priv=1;pw_info=operator;uid=2001;gid=100;group=users;dir=/home/operator;shell=/bin/rbash
+```
+
+the mapping of MPL would be as follows.
+
+| MPL (privilege)| user info      | uid  | gid  | secondary grps | shell     |
+| -------------- | -------------- | ---- | ---- | -------------- | --------- |
+| 15             | remote_user_su | 1000 | 1000 | sudo,docker    | /bin/bash |
+| 14-7           | netops         | 2007 |  100 | users          | /bin/rbash|
+| 1-6            | operator       | 2001 |  100 | users          | /bin/rbash|
+
+
+
+## ConfigDB Schema
+
+### AAA Table Schema
+
+```
+; Key
+aaa_key              = "authentication"   ; AAA type
+; Attributes
+login                = LIST(1*32VCHAR)   ; pam modules for particular protocol, now only support login for (local, tacacs+, radius). Legal combinations are [ [local], [local, tacacs+ ], [local, radius], [tacacs+, local], [radius, local] ]
+failthrough          = "True" / "False"  ; failthrough mechanism for pam modules
+debug                = "True" / "False"  ; debug logs for nss or pam modules
+```
+
+Note(s):
+- protocol = [[tacacs+, local], [radius, local]] are the options
+                                                 implementing fallback to local.
+- This is an existing table specification, which has been updated for RADIUS
+
+### RADIUS Table Schema
+
+```
+; RADIUS configuration attributes global to the system. Only one instance of the table exists in the system. Any of the global RADIUS attributes can be overwritten by the per server settings.
+; Key
+global_key           = "global"  ;  RADIUS global configuration
+; Attributes
+passkey              = 1*32VCHAR  ; shared secret (Valid chars: ASCII printable except SPACE, '#', and COMMA)
+auth_type            = "pap" / "chap" / "mschapv2"  ; method used for authenticating the communication message
+src_ip               = IPAddress  ;  source IP address (IPv4 or IPv6) for the
+                                  ;  outgoing UDP/IP dgram. This is being
+                                  ;  obsoleted in favor of the RADIUS_SERVER
+                                  ;  table src_intf attribute. If both src_ip
+                                  ;  and src_intf are specified a warning log
+                                  ;  can be given, and src_intf is preferred.
+                                  ;  Default is determined by routing stack.
+nas_ip               = IPAddress  ;  NAS-IP|IPV6-Address (Type 4|95) attribute
```

Comment:
What is the added value/gain/pro in configuring nas-ip?
Can you please provide some semantics on the usage ?
Some of it could get translated to help hints for user in CLI

Reply:
Every RADIUS PDU sent to the RADIUS server must have a NAS-IP|IPV6-Address or NAS-Identifier (or both) attribute. If the admin wants to 
(a) configure something other than the MGMT_INTERFACE table IP addresses
(b) reliably get one IP address (on a multi-homed/managementip device)

they could use this configuration.. Yes, I can add this to the HLD.

Reply:
This will be a very handy knob, to tune certain group of devices, differently. The knowledge in HLD doc will be very useful. Thanks

Reply:
Added additional nas-ip description to the RADIUS Table Schema in the form of a note.